### PR TITLE
Email Survey Button (Redmine #10684)

### PR DESCRIPTION
**Pull request by davidblader**
Email Survey button now deactivates when email fields do not match. Modal popup now appears. Added placeholder message within email text area.

### CODE REVIEW

**Comment by davidblader** [comment]
github isn't allowing me to add labels or milestones for some reason 💀 plz forgive me :bow:

**Comment by stellarxo** [comment]
@davidblader I gotchu

**Comment by davidblader** [comment]
@stellarxo thank u :simple_smile: 🙌 :clap:

**Comment by stellarxo** [comment]
I don't think the "Create Survey" should disable if an email is entered
image

**Comment by kongtiaowang** [comment]
screen shot 2016-10-04 at 2 45 28 pm
when it gets an error message, the email button should not work.

**Comment by stellarxo** [comment]
Duplicate error messages (one for create, one for email)
ALSO: survey instruments are not supposed to be part of the test battery, otherwise they would get populated automatically. There shouldn't be a check to see whether or not it's in the test battery - then a survey instrument would never be able to get added.

image

**Comment by stellarxo** [comment]
@davidblader still not able to add instruments that aren't part of the battery

image

**Comment by stellarxo** [comment]
@driusan can you please pass code review and merge?

File: modules/survey_accounts/js/survey_accounts_helper.js [hunk 1]
```diff
@@ -58,14 +57,36 @@ $(document).ready(function () {
...
+            $.get(loris.BaseURL + "/survey_accounts/ajax/ValidateEmailSubmitInput.php", {
+                dccid: $("input[name=CandID]").val(),
+                pscid: $("input[name=PSCID]").val(),
+                VL: $("select[name=VL]").val(),
+                TN: $("select[name=Test_name]").val(),
+                Email: $("input[name=Email").val(),
+                Email2: $("input[name=Email]").val()
+            },
+            function(result) {
+                //console.log(result);
```
Comment (stellarxo) [inline modules/survey_accounts/js/survey_accounts_helper.js]: Probably want to remove this

File: modules/survey_accounts/js/survey_accounts_helper.js [hunk 2]
```diff
@@ -58,14 +57,35 @@ $(document).ready(function () {
     });
     $("input[type=submit]").click(function (e) {
         if(e.currentTarget.classList.contains('email')) {
-            $("#emailModal").modal();
+            $.get(loris.BaseURL + "/survey_accounts/ajax/ValidateEmailSubmitInput.php", {
+                dccid: $("input[name=CandID]").val(),
```
Comment (alisterdev) [inline modules/survey_accounts/js/survey_accounts_helper.js]: @davidblader For future reference, to send the form data, all you have to do is something like that

```js
var formData = $('form[name=survey_accounts]').serialize();
$.get(loris.BaseURL + "/survey_accounts/ajax/ValidateEmailSubmitInput.php", data: formData);
```

File: modules/survey_accounts/js/survey_accounts_helper.js [hunk 3]
```diff
@@ -58,14 +57,35 @@ $(document).ready(function () {
...
+                pscid: $("input[name=PSCID]").val(),
+                VL: $("select[name=VL]").val(),
+                TN: $("select[name=Test_name]").val(),
+                Email: $("input[name=Email").val(),
+                Email2: $("input[name=Email]").val()
```
Comment (driusan) [inline modules/survey_accounts/js/survey_accounts_helper.js]: Shouldn't this be `input[name=Email2]`?

**Pull request merged**